### PR TITLE
Adds animal respawn for donators

### PR DESCRIPTION
**Pull request by Cruix**
Requested by Gray

Currently there are no limits on how many times you can do this, so that should probably be added.

### CODE REVIEW

**Comment by george99g** [comment]
Seems good. :+1:

File: code/modules/donors/donor_verbs.dm [hunk 1]
```diff
@@ -87,6 +88,31 @@ var/global/normal_donor_ooc_colour = "#333333"
...
+	for(var/mob/living/simple_animal/pet/P in living_mob_list)
+		if(!P.ckey && !P.stat)
+			options += P
+	var/mob/living/selected = input("Select a pet to become:") as null|anything in options
+	if(!selected || qdeleted(selected) || !istype(usr, /mob/dead/observer))
+		return
+	selected.ckey = ckey
+	usr.ckey = null
+
+	selected.pass_flags &= ~PASSDOOR
```
Comment (X-TheDark) [inline code/modules/donors/donor_verbs.dm:112]: Grrrr.

Comment (george99g) [inline code/modules/donors/donor_verbs.dm:112]: I don't think I understand what's supposed to happen here...
The flags get `AND`ed with the `NOT`ed `PASSDOOR`?

Comment (Cruix) [inline code/modules/donors/donor_verbs.dm:112]: It's to take away Ian's ability to go through doors, since that is easily exploitable.

Comment (X-TheDark) [inline code/modules/donors/donor_verbs.dm:112]: Yup, this is anti-Ian code, which is what I was making a jab at.

**Comment by Xantam-13** [comment]
Is this good to go?

**Comment by Cruix** [comment]
If you are happy with people being able to use it over and over it is

**Comment by Alek2ander** [comment]
What the fuck is wrong with you all?

**Comment by X-TheDark** [comment]
Yup, this is good to go.

**Comment by Kmc2000** [comment]
ADD 

YOU REMEMBER NO PAST LIVES if that wasnt already a thing.

Please

**Comment by Xantam-13** [comment]
Some notes for future dev on this;
- Make a little pop up display something along the lines of "Once you do this, you will lose any chance of returning to your human body"
- Only usable once (don't want to spam the station with a dozen dead dogs)
  -More I forgot.

**Comment by Cruix** [comment]
For the record, you don't become a new animal, you become one of the current living pets on the station, so there is a hard limit there unless someone goes crazy breeding corgis or the catsplosion happens.

**Comment by utahclock** [comment]
oh totes cool i can become roxie

**Comment by X-TheDark** [comment]
You know, this is a rather clever way of killing off the catsplosion cats. Too bad they won't be able to leave the CMO office on their own.

**Comment by ghost** [comment]
Bad idea, nerfs sentinence potioning ian, ruins immershun, possibility of metagame potential, poly can read himself wgw and then poly will remember and read wgw forever

**Comment by X-TheDark** [comment]
Paying to access this feature just so you can get yourself banned (mainly the WGW point) doesn't strike me as an idea that someone who decided to donate and support the server would do. Metagaming might be a slight issue, though.

**Comment by ShadowDeath6** [comment]
Poly can damage people so you might not want to have him due to possible griev and also teaching himself dumb shit

Also speaking of catsplosion can you just remove cat persistence please

**Pull request merged**